internal/pkg/aws/codepipeline/codepipeline_test.go

```diff
@@ -333,3 +333,126 @@ func TestCodePipeline_ListPipelinesForProject(t *testing.T) {
 		})
 	}
 }
+
+func TestCodePipeline_GetPipelineState(t *testing.T) {
+	mockPipelineName := "pipeline-dinder-badgoose-repo"
+	mockTime := time.Now()
+	mockOutput := &codepipeline.GetPipelineStateOutput{
+		PipelineName: aws.String(mockPipelineName),
+		StageStates: []*codepipeline.StageState{
+			{
+				ActionStates: []*codepipeline.ActionState{
+					{
+						LatestExecution: &codepipeline.ActionExecution{Status: aws.String(codepipeline.ActionExecutionStatusSucceeded)},
+					},
+				},
+				StageName: aws.String("Source"),
+			},
+			{
+				InboundTransitionState: &codepipeline.TransitionState{Enabled: aws.Bool(true)},
+				ActionStates: []*codepipeline.ActionState{
+					{
+						LatestExecution: &codepipeline.ActionExecution{Status: aws.String(codepipeline.ActionExecutionStatusInProgress)},
+					},
```

Comment:
In re: multiple action states -- my bad for missing that before, this would be a good place to add multiple actions per stage to make sure the behavior is correct!